keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -379,14 +390,14 @@ type IngressSpec struct {
 type IngressClass struct {
   metav1.TypeMeta
   metav1.ObjectMeta
-  Spec IngressClassSpec
-}
-
-type IngressClassSpec struct {
-  // This is currently empty.
 }
 ```
 
+The `IngressClass` object will be left empty for fields for the time
```

Comment:
I think it needs a single field which is the "provisioner" or "controller" responsible for handling a class.  Same as storage class.  We should probably also add the "Parameters" as a `TypedLocalObjectReference`.

E.g.

I could have classes:
* Gold = GCLB with params {tier=PREMIUM}
* Silver = GCLB with params {tier=STANDARD}
* Bronze = nginx

StorageClass under-specs the provisioner, but mostly it's assumed to be a label-key style string with domain prefix.

StorageClass specs the params as `map[string]string` but we have CRDs now.

Reply:
I can add this field as Tim suggests above, but wanted to be sure that this looks right first https://github.com/kubernetes/kubernetes/pull/80179 .

 When I view this file with the changes here it looks like it Class is of type string, but I'm not sure if that's intended. In my WIP PR I nest Class as type IngressClass under the IngressSpec.

Reply:
Adding 
```
// Controller is responsible for handling this class. This should be 
// specified as a domain-prefixed path, e.g. "acme.io/ingress-controller".
Controller string
// Parameters is a link to a customer resource configuration for the controller.
Parameters TypeLocalObjectReference
```